src/gcp-cli/orb.yml

```diff
@@ -0,0 +1,167 @@
+version: 2.1
+description: |
+  Installs the Google Cloud Command Line Interface (gcloud)
+
+  The executors within this Orb provide run-time environments that contain the
+  necessary dependencies for executing the jobs and commands in this orb,
+  except for the following required dependencies that must be configured in
+  CircleCI in order to use this orb - environment variables for gcloud
+  initialization ($GCLOUD_SERVICE_KEY).
+
+  The source for this Orb can be found here:
+  https://github.com/CircleCI-Public/circleci-orbs/tree/master/src/gcp-cli
+
+examples:
+  simple_install_and_configure:
+    description: Install the gcloud CLI, if not available
+    usage:
+      version: 2.1
+
+      orbs:
+        gcp-cli: circleci/gcp-cli@1.0.0
+
+      workflows:
+        install_and_configure_cli:
+          jobs:
+            - gcp-cli/install_and_configure_cli:
+                context: myContext # your context containing gcloud service key
+                google-project-id: myGoogleProjectId
+                google-compute-zone: myGoogleComputeZone
+
+executors:
+  default:
+    description: A debian based docker container to use when running the
+                 gcloud CLI
+    parameters:
+      python-version:
+        type: string
+        default: "2.7"
+      debian-release:
+        type: string
+        default: "stretch"
+    docker:
+      - image: circleci/python:<< parameters.python-version >>-<< parameters.debian-release >>
+  google:
+    description: The official Google docker container with gcloud SDK
+                 pre-installed
+    docker:
+      - image: google/cloud-sdk
+
+commands:
+  install:
+    description: |
+      Install the gcloud CLI, if not available
+    steps:
+      - when:
+          name: Install gcloud CLI, if not available
+          condition: $(command -v gcloud) == ""
```

Comment:
Does this work? I  thought conditionals were evaluated before instantiating any containers.
>These conditions are checked before a workflow is actually run. That means, for example, that a user can't use a condition to check an environment variable.